### PR TITLE
Feature/mixer-10: Get Youtube title from API

### LINKED ISSUE #10: Get YouTube video title from API

### PR DESCRIPTION
**Pull request by DijitGuy**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,3 +49,4 @@ testem.log
 # System Files
 .DS_Store
 Thumbs.db
+src/environments/Keys.ts
```
Comment (Froopity) [inline .gitignore]: Already exists under environment variables

File: src/app/app.module.ts [hunk 2]
```diff
@@ -37,6 +38,7 @@ import { CrossfadeButtonsComponent } from './component/crossfade-buttons/crossfa
     FormsModule,
     MatFormFieldModule,
     MatInputModule,
+    HttpClientModule,
```
Comment (Froopity) [inline src/app/app.module.ts]: I try to keep these alphabetically sorted

File: src/app/component/youtube-video/youtube-video.component.ts [hunk 3]
```diff
@@ -39,9 +40,10 @@ export class YoutubeVideoComponent implements OnInit, OnDestroy {
   private readonly id: AbstractControl;
 
   videoId$ = new BehaviorSubject<string>('oHg5SJYRHA0');
+  videoTitle$ = new BehaviorSubject<string>('RickRoll\'D');
```
Comment (Froopity) [inline src/app/component/youtube-video/youtube-video.component.ts]: You can use backticks ( \` ) to escape entire strings. Not necessary here but useful to know. Using backticks also lets you put expressions straight into strings e.g. `` `Number: ${numVar}` ``

File: src/app/component/youtube-video/youtube-video.component.ts [hunk 4]
```diff
@@ -39,9 +40,10 @@ export class YoutubeVideoComponent implements OnInit, OnDestroy {
...
 
   constructor(private readonly volumeService: VolumeService,
-              formBuilder: FormBuilder) {
+              formBuilder: FormBuilder, private youtubeService: YoutubeService) {
```
Comment (Froopity) [inline src/app/component/youtube-video/youtube-video.component.ts]: I try to put fields above other parameters in constructors. Also readonly, so:
``` 
constructor(private readonly volumeService: VolumeService,
            private readonly youtubeService: YoutubeService,
            formBuilder: FormBuilder) {

File: src/app/component/youtube-video/youtube-video.component.ts [hunk 5]
```diff
@@ -86,5 +88,11 @@ export class YoutubeVideoComponent implements OnInit, OnDestroy {
     const values = value.split('=');
 
     this.videoId$.next(values[values.length - 1]);
+
+    this.youtubeService.getDataForVideo(values[values.length - 1]).subscribe(
```
Comment (Froopity) [inline src/app/component/youtube-video/youtube-video.component.ts]:
```
this.youtubeService.getDataForVideo(values[values.length - 1]).subscribe(data =>
    this.videoTitle$.next(data.items[0].snippet.title)
);

File: src/app/service/youtube.service.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {Keys} from '../../environments/keys';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class YoutubeService {
+
+  apiKey = Keys.YOUTUBE_API_KEY;
```
Comment (Froopity) [inline src/app/service/youtube.service.ts]: private readonly

File: src/app/service/youtube.service.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {Keys} from '../../environments/keys';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class YoutubeService {
```
Comment (Froopity) [inline src/app/service/youtube.service.ts]: Maybe YoutubeApiService?

File: src/app/service/youtube.service.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+  apiKey = Keys.YOUTUBE_API_KEY;
+
+  constructor(public http: HttpClient) { }
```
Comment (Froopity) [inline src/app/service/youtube.service.ts:13]: private readonly

File: src/app/service/youtube.service.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+
+  getDataForVideo(video: string): Observable<any> {
+    const url = 'https://www.googleapis.com/youtube/v3/videos?id=' + video + '&key=' + this.apiKey + '&part=snippet,statistics';
+    return this.http.get(url);
+  }
+
+  extractData(res: Response): object {
+    const body = res;
+    return body || {};
+  }
```
Comment (Froopity) [inline src/app/service/youtube.service.ts]: This doesn't do anything right? Can probably go

**Review by Froopity — CHANGES_REQUESTED** [review]

File: src/app/component/youtube-video/youtube-video.component.ts (hunk 3 above)
Comment (DijitGuy) [reply src/app/component/youtube-video/youtube-video.component.ts]: thats sick actually, thanks

**Review by Froopity — APPROVED** [review]

**Pull request merged**